src/KnowbaseItem.php

```diff
@@ -888,7 +903,23 @@ public function showFull($options = [])
          */
         global $CFG_GLPI, $DB;
 
-        if (!$this->can($this->fields['id'], READ)) {
+        if (
+            $this->fields['is_token_url']
+            && isset($options['token'])
+            && $this->fields['token'] == $options['token']
+        ) {
+            $answer = preg_replace(
+                [
+                    '/<img([^>]+src=["\'])([^"\']+)(["\'][^>]*)>/',
+                    '/<a([^>]+href=["\'])([^"\']+)(["\'][^>]*)>/'
```

Comment:
Can you explain why these regex are needed and what they are doing ? I suppose it add `&token=xxx` to images and links inside the article to allow them to be visible ?

Shouldn't it be done only for urls to /front/document.send.php ? And how does it validate the token ?

Reply:
> Can you explain why these regex are needed and what they are doing ? I suppose it add `&token=xxx` to images and links inside the article to allow them to be visible ?

Yes.

> Shouldn't it be done only for urls to /front/document.send.php ? And how does it validate the token ?

It's `KnowbaseItem->canViewItem()`